### ISSUE TITLE
pnpm list only dev/prod not working

### ISSUE THREAD

**Issue by andretf** [pnpm/pnpm#1216]
My `pnpm ls` is not restricting to environment dependencies, if I use `--prod`, `--dev`, or one of their aliases. For both flags, it prints all installed dependencies.

### pnpm version:
2.4.0

### Code to reproduce the issue:
    $ pnpm ls --prod --depth 0

### Expected behavior:
    ├── del@3.0.0

### Actual behavior:
```
├── codeclimate-test-reporter@0.5.0
├── del@3.0.0
├── gulp@3.9.1
├── gulp-concat@2.6.1
├── gulp-jasmine-phantom@3.0.0
├── gulp-rename@1.3.0
├── gulp-size@2.1.0
├── gulp-uglify@1.5.4
├── jasmine-core@3.1.0
├── karma@2.0.2
├── karma-chrome-launcher@2.2.0
├── karma-coverage@1.1.2
├── karma-firefox-launcher@1.1.0
├── karma-ie-launcher@1.0.0
├── karma-jasmine@1.1.2
├── karma-opera-launcher@1.0.0
├── karma-phantomjs-launcher@1.0.4
├── karma-safari-launcher@1.0.0
└── phantomjs-prebuilt@2.1.16
```

### Additional information:
shrinkwrap.yaml:
```
dependencies:
  del: 3.0.0
devDependencies:
  codeclimate-test-reporter: 0.5.0
  gulp: 3.9.1
  gulp-concat: 2.6.1
  gulp-jasmine-phantom: 3.0.0
  gulp-rename: 1.3.0
  gulp-size: 2.1.0
  gulp-uglify: 1.5.4
  jasmine-core: 3.1.0
  karma: 2.0.2
  karma-chrome-launcher: 2.2.0
  karma-coverage: 1.1.2
  karma-firefox-launcher: 1.1.0
  karma-ie-launcher: 1.0.0
  karma-jasmine: 1.1.2
  karma-opera-launcher: 1.0.0
  karma-phantomjs-launcher: 1.0.4
  karma-safari-launcher: 1.0.0
  phantomjs-prebuilt: 2.1.16
...
```

package.json:
```
  "devDependencies": {
    "codeclimate-test-reporter": "0.5.0",
    "gulp": "^3.9.1",
    "gulp-concat": "^2.6.1",
    "gulp-jasmine-phantom": "^3.0.0",
    "gulp-rename": "^1.2.2",
    "gulp-size": "^2.0.0",
    "gulp-uglify": "^1.5.2",
    "jasmine-core": "^3.1.0",
    "karma": "^2.0.2",
    "karma-chrome-launcher": "^2.2.0",
    "karma-coverage": "^1.1.2",
    "karma-firefox-launcher": "^1.1.0",
    "karma-ie-launcher": "^1.0.0",
    "karma-jasmine": "^1.1.2",
    "karma-opera-launcher": "^1.0.0",
    "karma-phantomjs-launcher": "^1.0.4",
    "karma-safari-launcher": "^1.0.0",
    "phantomjs-prebuilt": "^2.1.5"
  },
  "dependencies": {
    "del": "3.0.0"
  }
```

 - `node -v` prints: v10.3.0
 - Windows, OS X, or Linux?: Linux Manjaro

**Comment by andretf** [comment]
Ok, I got what is happening.

**TLDR:**

This works: `pnpm --only [dev|prod]`

---
This is what I get for `pnpm ls --help`:

```
pnpm ls [[<@scope>/]<pkg> ...]

Aliases: list, la, ll

When run as ll or la, it shows extended information by default.

Options:

  --long                          show extended information
  --parseable                     show parseable output instead of tree view
  -g, --global                    list packages in the global install prefix instead of in the current project
  --depth                         max display depth of the dependency tree
  --prod, --production            display only the dependency tree for packages in `dependencies`.
  --dev                           display only the dependency tree for packages in `devDependencies`.
```

Looking at [source code](https://github.com/pnpm/pnpm/blob/master/packages/pnpm/src/cmd/list.ts#L11) it should be something like:
```
  --only prod            display only the dependency tree for packages in `dependencies`.
  --only dev             display only the dependency tree for packages in `devDependencies`.
```
..or source code fixed.

**Comment by zkochan** [comment]
Source code should be fixed. I thought `@pnpm/config` would normalize these flags https://github.com/pnpm/pnpm/tree/master/packages/config

**Comment by zkochan** [comment]
:ship: 2.6.0
